v7/_posts/odata features/2017-12-26-06-00-JsonBatch.md

```diff
@@ -0,0 +1,330 @@
+---
+title: "Json batch format in ODataLib"
+layout: post
+category: "6. OData Features"
+description: ""
+---
+
+# Introduction to JSON Batching 
+
+## Overview
+Similar to Windows batch command, OData supports grouping multiple requests together, sending a single HTTP request to OData service, and grouping corresponding responses together before sending back as a single HTTP response. The major benefits for request batching are to reduce client/server round trips and to support atomic operation.
+
+The batch format supported by OData core libraries is multipart/mime for OData protocol up to v4.0. To accommodate the need for a more developer-friendly format, the new JSON format batching support is added to the latest version of OData protocol v4.01. The JSON format batching format also brings another major benefit of allowing requests inside a batch to be executed in specified orders.
+
+Draft iterations of OData protocol v4.01 can be found on oasis site. At the time of this writing the latest draft version is Draft 04: http://docs.oasis-open.org/odata/odata/v4.01/csprd04/part1-protocol/odata-v4.01-csprd04-part1-protocol.html#_Toc499805328.
+
+Draft iterations of OData JSON format v4.01 can also be found on oasis site. At time of this writing the latest JSON Batch format is Draft 04: http://docs.oasis-open.org/odata/odata-json-format/v4.01/csprd04/odata-json-format-v4.01-csprd04.html#_Toc499716905
+
+## Sample Batch Request in JSON Format
+Here is one sample batch request in JSON format (unnecessary details are omitted for sake of brevity):
+
+{% highlight csharp %}    
```

Comment:
highlighting of body should be json, not csharp, and md doesn't seem to recognize your "highlight" directive.  try replacing "{% highlight csharp %}" with "~~~json" and "{% endhighlight %}" with "~~~".  i.e.;

~~~json

{
   "requests" : [
...
   ]
}
~~~

Reply:
Updated.